tmt/utils.py

```diff
@@ -4787,11 +4796,18 @@ def git_clone(
                 url, destination
                 ), env=env)
     except RunError:
+        if retry_times == 0:
+            raise
         if not shallow:
             # Do not retry if shallow was not used
             raise
         # Git server might not support shallow cloning, try again (do not modify url)
-        return git_clone(url, destination, common, env, shallow=False, can_change=False)
+        shallow = retry_times > 1
+        retry_times -= 1
+        return git_clone(url, destination, common, env,
+                         shallow=shallow,
+                         can_change=False,
```

Comment:
Why is `can_change` modified? If the function is called with `can_change=True`, is retrying expected to change this value?

Reply:
Hi @happz, let me explain more,

a) If the function is called with `can_change=True` at the first time，

```python
4792     if can_change:
4793         url = clonable_git_url(url)
```

The `url` is changed. So I think if the function is called at the second time, the third time, ..., it is not necessary to change the `url`.

b）If the function is called with `can_change=False` at the first time, `url` will not be changed. So the `url` is always the origin one when the function is re-called.

What do you think?

Reply:
Makes sense, thanks. Please, add the shortened version fo this explanation as a comment, it will be useful for the future readers :)

Reply:
This seems to be addressed in the comment.